config/datastore.go

```diff
@@ -21,8 +34,10 @@ type Datastore struct {
 
 	Spec map[string]interface{}
 
-	HashOnRead      bool
-	BloomFilterSize int
+	HashOnRead        bool
+	BloomFilterSize   int
+	BlockKeyCacheSize OptionalInteger `json:",omitempty"`
+	WriteThrough      bool
```

Comment:
@hsanjuan are you planning to write a migration for existing users (who don't have this in config), or was it intentional to keep existing users stuck with `false`? 

Both feel like a potential headache :)

Perhaps we should switch this from `bool` to `Flag` would allow us to control implicit default better, without having to write migration for existing users?

```suggestion
	WriteThrough      Flag `json:",omitempty"`
```

Reply:
If the option is not present, it will take the default which is true? Or you mean the config-parsing code does not know if its false or unset?